src/python/WMCore/Storage/RucioFileCatalog.py

```diff
@@ -0,0 +1,236 @@
+        #it is a site (no defined subSite), storage.json is located at the path given in SITECONFIG_PATH
+        if currentSubsite is None:
+            subPath = siteConfigPath
+        #it is a subsite, move one level up
+        else:
+            subPath = siteConfigPath + '/..'
+    #cross site
+    else:
+        #it is a site (no defined subSite), move one level up
+        if currentSubsite is None:
+            subPath = siteConfigPath + '/../' + storageSite
+        #it is a subsite, move two levels up
+        else:
+            subPath = siteConfigPath + '/../../' + storageSite
+    pathToStorageDescription = subPath + '/storage.json'
+    pathToStorageDescription = os.path.normpath(os.path.realpath(pathToStorageDescription))#resolve symbolic link and relative path?
+    return pathToStorageDescription 
+
+def readRFC(filename, storageSite, volume, protocol):
+    """
+    _readRFC_
+
+    Read the provided storage.json and return a RucioFileCatalog
+    instance containing the details found in it
+    """
+    rfcInstance = RucioFileCatalog()
+    try:
+        jsonFile = open(filename,encoding="utf-8")
+        jsElements = json.load(jsonFile)
+    except Exception as ex:
+        msg = "Error reading storage description file: %s\n" % filename
+        msg += str(ex)
+        raise RuntimeError(msg)
+    #now loop over elements, select the right one and fill lfn-to-pfn
+    for jsElement in jsElements:
+        #check to see if the storageSite and volume matchs with "site" and "volume" in storage.json
+        if jsElement['site'] == storageSite and jsElement['volume'] == volume: 
+            rfcInstance.preferredProtocol = protocol
+            #now loop over protocols to add all mappings (needed for chained rule cases)
+            for prot in jsElement['protocols']:
+                #check if prefix is in protocol block
+                if 'prefix' in prot.keys():
+                    #lfn-to-pfn
+                    match = '/(.*)' #match all
+                    result = prot['prefix']+'/$1'
+                    #prefix case should not be chained
+                    chain = None 
+                    rfcInstance.addMapping(str(prot['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                    #pfn-to-lfn
+                    match = prot['prefix']+'/(.*)'
+                    result = '/$1'
+                    rfcInstance.addMapping(str(prot['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+                #here is rules  
+                else:
+                    #loop over rules
+                    for rule in prot['rules']:
+                        match = rule['lfn']
+                        result = rule['pfn']
+                        chain = rule['chain'] if 'chain' in rule.keys() else None
+                        rfcInstance.addMapping(str(prot['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                        #pfn-to-lfn: not sure about this!!!
```

Comment:
About what exactly? Whether we need to replace strings here? Should we contact someone else to be certain about this?

Reply:
Yes, I attack the most common strings used in lfn currently (`replace('/+','/').replace('^/','/')`). As far as I know the `pfn-to-lfn` has not been used anymore, so this issue is not important. We can keep as is for now. I update comment:
"#Update this if pfn-to-lfn is used extensively somewhere. We want lfn starts with '/abc' so remove all characters of regular expressions!!! "